h2o-algos/src/main/java/hex/ensemble/StackedEnsemble.java

```diff
@@ -66,6 +66,33 @@ public boolean haveMojo() {
     return true;
   }
 
+  @Override
+  public void init(boolean expensive) {
+    super.init(expensive);
+
+    checkFoldColumnPresent(_parms._metalearner_fold_column, _parms._train != null ? _parms._train.get() : null,
```

Comment:
This is IMHO not entirely correct. You are not actually validating the frame that the algorithm will use, you are validating the original raw (unadapted) frame.

I think at this point you should be using the `train()`/`valid()` frames otherwise the init fail will pass but algo will fail when it tries to train the metalearner (it will not see any fold column and will fail on validation).

I added a commit with a failing test that demonstrates that.

Reply:
Oh, I dod not realize I'm not using adapted frames. :space_invader: So I did fix this for validation & training dataframe. Blending frame is taken as-is from the params, as the blending driver does the same.

You can see the changes in https://github.com/h2oai/h2o-3/pull/3592/commits/7a13e95c7c1d341653727eee10beb8ab9d34cc71

Please note I've deleted the frame key from the error mesasge, as the adapated temporary frames do not have a key and thus there would be null for train/valid.